theme/management/commands/report_resource_size_vs_quota_inconsistency.py

```diff
@@ -0,0 +1,189 @@
+                    res_files = []
+                    updated_size = 0
+                    for res in held_resources:
+                        print(f"Total files in resource {res.short_id}: {res.files.all().count()}")
+                        print(f'{current_site}/resource/{res.short_id}: currently {res.size} bytes')
+                        file_counter = 0
+                        # exclude files with size 0 as they don't exist in iRODS
+                        for res_file in res.files.exclude(_size=0).iterator():
+                            # this is an expensive operation (3 irods calls per file) - about 1 min for 100 files
+                            # size, checksum and modified time are obtained from irods and assigned to
+                            # relevant fields of the resource file object
+                            res_file.set_system_metadata(resource=res, save=False)
+                            res_files.append(res_file)
+                            file_counter += 1
+                            print(f"Updated file count: {file_counter}")
+                            if res_file._size <= 0:
+                                print(f"File {res_file.short_path} was not found in iRODS.")
+
+                        if res_files:
+                            ResourceFile.objects.bulk_update(res_files,
+                                                             ResourceFile.system_meta_fields(), batch_size=_BATCH_SIZE)
+                            print(f"Updated {file_counter} files for resource {res.short_id}")
+                            res.refresh_from_db()
+                            print(f'{current_site}/resource/{res.short_id}: now {res.size} bytes')
+                            # keep track of the updated size so that we can compare again
+                            updated_size += res.size
+                        else:
+                            print(f"Resource {res.short_id} contains no files.")
+                    converted_updated_size_django = convert_file_size_to_unit(int(updated_size), 'gb')
+                    report_dict['django_updated'] = converted_updated_size_django
+                    if not math.isclose(used_value_irods_dz, converted_updated_size_django, abs_tol=0.1):
+                        print("Even after updating, an inconsistency remains!")
+                        print(f"Quota usage in iRODS Datazone: {used_value_irods_dz} GB")
+                        print(f"Updated Total size of resources in Django: {converted_updated_size_django} GB")
+
+                elif reset:
+                    print("Resetting file size cache in django")
+                    for res in held_resources:
+                        print(f'Resetting all filesizes in {current_site}/resource/{res.short_id}')
+                        res.files.update(_size=-1)
+                        res.updated = now().isoformat()
```

Comment:
We also need to update the modified date resource metadata?
Ref: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/utils.py#L618

Reply:
thanks @pkdash! Updated to use `resource_modified`

Reply:
scratch that, I forgot that admins cant use `resource_modified` so I will fix